tools/topology/sof-tgl-rt711-rt1308.m4

```diff
@@ -20,22 +20,45 @@ include(`common/tlv.m4')
 # Include Token library
 include(`sof/tokens.m4')
 
-include(`platform/intel/tgl.m4')
+include(`platform/intel/'PLATFORM`.m4')
 
 # Define pipeline id for intel-generic-dmic.m4
 # to generate dmic setting
-
-ifelse(CHANNELS, `0', ,
+ifdef(`CHANNELS',
```

Comment:
does this work? ifdef and ifelse are different. What you here is a value that's explicitly not zero.

Reply:
@plbossart 
Both works, but ```ifelse(CHANNELS, `0', ,...``` is more complicated and easy wrongly used. The distinguish is:
If we use ```ifelse(CHANNELS, `0', ,...```, we have to set -DCHANNELS=0 in CMakefiles.txt. Otherwise, CHANNELS is not '0', but not defined. Please check https://github.com/thesofproject/sof/blob/5ca0e9bba3e6da8abab7faf4577aa6683200bf41/tools/topology/CMakeLists.txt#L23 for the correct usage. However, developer may forget to set  -DCHANNELS=0 if he doesn't use the DMIC, which will lead to wrong result.
If we use ```ifdef(`CHANNELS', ...```, we needn't set -DCHANNELS=0 in CMakefiles.txt, which is more easy for us.

Please check the https://www.gnu.org/savannah-checkouts/gnu/m4/manual/m4-1.4.18/m4.html#Ifdef for the difference between ifdef and ifelse.

Reply:
I disagree. The current usage is to explicitly set CHANNELS=0 in CMakeFiles.txt. That's the convention that was used before, let's follow it and not change the expectations just for one topology.

When there are pre-existing conventions, we follow them, even if they are not necessarily required. Having ONE pattern makes it easier to maintain those files.

````
"sof-hda-generic\;sof-hda-generic\;-DCHANNELS=0\;-DHSPROC=volume\;-DDYNAMIC=1"
"sof-hda-generic-idisp\;sof-hda-generic-idisp\;-DCHANNELS=0\;-DDYNAMIC=1"
````

Reply:
@plbossart Got it. I will update the patch.